bittensor/_neuron/text/template_server/nucleus_impl.py

```diff
@@ -131,34 +132,39 @@ def encode_forward(self,inputs,tokenizer=None):
                     The tokenizer which was used to tokenize the inputs
 
             Returns:
-                outputs (:obj:`torch.FloatTensor`):
-                    The nucleus's outputs as a torch tensor of shape [batch_size, sequence_len, __network_dim__]
+                model_outputs (:type:`SimpleNamespace`[logits (:obj:`torch.tensor`), hidden (:obj:`torch.tensor`)], `required`):
+                    logits: The logits output as a torch tensor of shape [batch_size, sequence_len, __vocab_size__ ]
+                    hidden: The hidden layer output as a torch tensor of shape [batch_size, sequence_len, __network_dim__ ]
         """
         sen_len = inputs.size()
         inputs = self.token_remap(inputs,tokenizer).to(self.device)
-        if self.config.neuron.training:
-            pre_hidden = self.pre_model(inputs).last_hidden_state
-        elif self.config.neuron.autocast and self.device == 'cuda':
-            pre_hidden = self.pre_model(inputs).last_hidden_state
+        if self.config.neuron.training or self.config.neuron.autocast and self.device == 'cuda':
+            output = self.pre_model(inputs, output_hidden_states=True)
+            pre_hidden = output.hidden_states[-1]
         else:
             with torch.no_grad():
-                pre_hidden = self.pre_model(inputs).last_hidden_state
-        
+                output = self.pre_model(inputs, output_hidden_states=True)
```

Comment:
Would this always compute the logits layer as well? If so, do we know how much time the last layer takes?

Reply:
pre_model() always compute the whole model, always including the last layer, setting option output_hidden_states = True just add a return variable.